### PR TITLE
Notebooks swagger spec

### PR DESCRIPTION
**Pull request by somu84**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by erich-wang** [comment]
@somu84 , could you please fix the validation errors in CI check?

**Comment by chiragg4u** [comment]
This looks like a completely new RP, please setup meeting with the API review board and review the scenario and API model.

File: specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+                  "type": "string"
+              },
+              {
+                  "name": "resourceName",
+                  "in": "path",
+                  "required": true,
+                  "type": "string"
+              }
+          ],
+          "responses": { "200" : { "description": "Success" } }
```
Comment (chiragg4u) [inline specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json:35]: >"description": "Success" [](start = 35, length = 24)

This should return the resource.

File: specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/example/NBResourceCreate.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "parameters": {
+    "resourceName": "testResource",
+    "resourceGroupName": "testRP",
+    "subscriptionId": "91e9f7ac-9c2c-4cd2-af61-1c5b57ec6c51",
+    "resourceReuest": {
+      "location": "southcentralus",
+      "properties": {
+      },
```
Comment (chiragg4u) [inline specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/example/NBResourceCreate.json:9]: you need to pass the properties in PUT

File: specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/example/NBResourceUpdate.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "parameters": {
+    "resourceName": "testResource",
+    "resourceGroupName": "testRP",
+    "subscriptionId": "91e9f7ac-9c2c-4cd2-af61-1c5b57ec6c51"
+  },
```
Comment (chiragg4u) [inline specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/example/NBResourceUpdate.json:6]: What is the update?

**Comment by somu84** [comment]
> This looks like a completely new RP, please setup meeting with the API review board and review the scenario and API model.

Hi Chirag, we are already on-boarded to ARM. [This](https://github.com/Azure/azure-rest-api-specs-pr/pull/1086) is the PR that was approved by ARM team last year. This is a top level proxy resource. We want it to be hidden from end customers.

File: specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/example/NBResourceCreate.json (hunk 2 above)
Comment (somu84) [reply specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/example/NBResourceCreate.json:9]: I have linked a prior[ PR](https://github.com/Azure/azure-rest-api-specs-pr/pull/1086) that has been approved.

**Comment by somu84** [comment]
> @somu84 , could you please fix the validation errors in CI check?

Yes, I will push the fixes to the comments today.

File: specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+            "200": { 
+              "description": "Success" 
+            }, 
+            "204": {
+              "description": "Success"
+            } 
+          }
+        }
+      },
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Notebooks/NotebookProxies/{resourceName}/updatePrivateLinkId/{privateLinkId}": {
```
Comment (ravbhatnagar) [inline specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json:112]: what is this API supposed to do?

File: specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+            },
+            {
+              "name": "resourceRequest",
+              "in": "body",
+              "required": false,
+              "schema": { "$ref": "#/definitions/NotebookResource" }
+            }
+          ],
+          "responses": { 
+            "200": {
```
Comment (ravbhatnagar) [inline specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json:69]: response body missing

File: specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+        }
+      }
+    },
+    "definitions": {
+      "NotebookResource": {
+        "type": "object",
+        "properties": {
+          "location": { "type": "string" },
+          "properties": { "$ref": "#/definitions/NotebookResourceProperties" },
+          "hostname": { "type": "string" }
```
Comment (ravbhatnagar) [inline specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json:183]: this cant be a top level property, it must be moved under "properties" property

**Review by ravbhatnagar — COMMENTED** [review]
@somu84 few comments

File: specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json (hunk 4 above)
Comment (somu84) [reply specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json:112]: This API is updating the resource by adding a private link Id.

**Comment by erich-wang** [comment]
@somu84, please fix validation error, you may refer to https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md

**Comment by erich-wang** [comment]
@somu84, is this PR still valid? I'll close this PR if it is not, thanks.

**Comment by somu84** [comment]
> @somu84, is this PR still valid? I'll close this PR if it is not, thanks.

Sorry, I was pulled into other things. I have checked-in the fixes for the CI errors.

**Comment by erich-wang** [comment]
@somu84 , have you addressed the comments raised by @ravbhatnagar?

File: specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json (hunk 5 above)
Comment (somu84) [reply specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json]: Fixed

File: specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json (hunk 6 above)
Comment (somu84) [reply specification/notebooks/resource-manager/Microsoft.Notebooks/preview/2019-10-11-preview/notebooks.json]: Fixed

**Comment by somu84** [comment]
> @somu84 , have you addressed the comments raised by @ravbhatnagar?

Yes, I have addressed the comments from @ravbhatnagar

**Comment by somu84** [comment]
@erich-wang please let me know, if there are any blocking issues, that I need to resolve?

**Comment by erich-wang** [comment]
@ravbhatnagar, do you have any other comments? Could you please sign off this PR ?

**Comment by somu84** [comment]
Any update on this?

**Comment by ravbhatnagar** [comment]
signing off from ARM side.

**Comment by erich-wang** [comment]
@somu84, please remove `WaitForARMFeedback` from the title since it is approved by ARM review board, then I'll merge this PR.

**Review by erich-wang — APPROVED** [review]

**Comment by somu84** [comment]
@erich-wang done!

**Pull request merged**